### ISSUE TITLE
UI error - contradictory output when toolchain not found

### ISSUE THREAD

**Issue by boozook** [rust-lang/rustup#3944]
### Verification

- [X] I searched for recent similar issues at https://github.com/rust-lang/rustup/issues?q=is%3Aissue+is%3Aopen%2Cclosed and found no duplicates.
- [X] I am on the latest version of Rustup according to https://github.com/rust-lang/rustup/tags and am still able to reproduce my issue.

### Problem

Unexpected output:

```
info: no toolchain installed for 'nightly-2023'
info: toolchain 'nightly-2023' uninstalled
```

### Steps

1. `rustup install nightly-2022-10-19-aarch64-apple-darwin` (actually __any__ toolchain you want)
2. `rustup uninstall nightly-2022`
3. read the contradictory output

Expected output can be only one -
`no toolchain installed for 'nightly-2022'` __or__ `toolchain 'nightly-2022' uninstalled`

### Rustup version

```console
rustup 1.27.1 (54dd3d00f 2024-04-24)
```

### Installed toolchains

```console
It doesn't matter. Many nightly, zero or one stable.
```

### OS version

```console
macOS 14.5
```

**Comment by rami3l** [comment]
@boozook Thanks for filing this issue! I believe this has been fixed by #3869.

**Comment by boozook** [comment]
Oops 😬 
Sorry for the double.
